### PR TITLE
Email Smtp Setup

### LINKED ISSUE #75: Setup SMTP service provider [Auto Email]

### PR DESCRIPTION
**Pull request by zollf**
Closes #75

I've added a model to keep track of email events. This is more to keep track of emails somewhere in the db, the views are just extra.

## Proposed Changes
  - Added emails page `admin/emails`
  - Added email page `admin/emails/view/<int:pk>`
  - Added API endpoint for email testing `api/emails/test`
  - Added model for keeping track of email events if needed
  - Added .env variables 
  - Added env variables to `deploy.yml`

## TODO:
- currently, I've limited emails to 10 on the page so loading everything isn't an issue (it starts to be an issue in the 1000s usually), if someone has spare time they can do pagination. However its not a big deal as its logged in db which is more important.

### CODE REVIEW

**Comment by APM246** [comment]
I'll try to review this tomorrow (latest by Monday)

File: app/emails/send.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    )
+    msg.attach_alternative(html_body, "text/html")
+    msg.send(fail_silently=False)
+
+    # If msg.send does not fail then we log
+    email = Emails(
+        name=Emails.events[category]['name'],
+        description=Emails.events[category]['description'],
+        category=category,
+        payload="",
```
Comment (APM246) [inline app/emails/send.py]: Shouldn't this be payload = html_body?

File: app/emails/send.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from django.core.mail import send_mail, EmailMultiAlternatives
```
Comment (APM246) [inline app/emails/send.py]: remove unused import send_email

**Review by APM246 — CHANGES_REQUESTED** [review]

File: app/emails/send.py (hunk 1 above)
Comment (zollf) [reply app/emails/send.py]: its supposed to be data actually, i need to change that

**Review by APM246 — APPROVED** [review]

**Pull request merged**